lib/kano-register/kano-email-field.js

```diff
@@ -0,0 +1,96 @@
+var xtag = window.xtag;
+
+module.exports = function () {
+    xtag.register('kano-email-field', {
+        content: function () {/*
+            <div class="kano-email-field">
+                <label>Parent's email address</label>
+                <input id="input" class='block' type='email' autofocus placeholder='E.g. johndoe@gmail.com' required />
+                <div id="errors" class="errors"></div>
+            </div>
+        */},
+
+        accessors: {
+            value: {
+                attribute: {},
+                get: function () {
+                    return this.data.value;
+                },
+                set: function (val) {
+                    this.valueSetter(val, true);
+                }
+            },
+            silentValidation: {
+                attribute: {},
+                set: function (val) {
+                    this.data.silentValidation = (val === "true");
+                    this.showErrors(!this.data.silentValidation);
+                },
+                get: function () {
+                    return this.data.silentValidation;
+                }
+            }
+        },
+
+        lifecycle: {
+            created: function () {
+                this.data = {};
+                this.input = xtag.query(this, '#input')[0];
+                this.errors = xtag.query(this, '#errors')[0];
+                this.isValid = false;
+                this.pristine = true;
+            }
+        },
+
+        events: {
+            keyup: function () {
+                clearTimeout(this.data.timeout);
+                this.data.timeout = setTimeout(this.valueSetter.bind(this, this.input.value), 200);
+            }
+        },
+
+        methods: {
+            valueSetter: function (val, fromOutside) {
+                var errors = this.getFormatErrors(val),
+                    errorMessage;
+                this.pristine = false;
+                if (!errors) {
+                    this.data.value = val;
+                    this.displayError();
+                    this.isValid = true;
+                    if (fromOutside) {
+                        this.input.value = val;
+                    }
+                } else {
+                    errorMessage = errors.reduce(function (acc, item) {
+                        return acc + '<li>' + item + '</li>';
+                    }, '<b>Email must</b><ul>');
```

Comment:
There is no real template with x-tag, only content. so everything must be done through DOM manipulation. I guess we could do something better by creating the ul and li  and append it instead of setting the innerHTML